mutiny-core/src/lib.rs

```diff
@@ -1363,6 +1377,85 @@ impl<S: MutinyStorage> MutinyWallet<S> {
         });
     }
 
+    /// Creates a node manager with a node and returns the node identity.
+    /// If a node manager already exists, it will return an error.
+    pub async fn create_node_manager(&self) -> Result<NodeIdentity, MutinyError> {
+        if self.is_safe_mode() {
+            return Err(MutinyError::NotRunning);
+        }
+
+        let mut lock = self.node_manager.write().await;
+        // if we already have a node manager, return
+        if lock.is_some() {
+            return Err(MutinyError::AlreadyRunning);
+        }
+
+        let mut nm_builder = NodeManagerBuilder::new(self.xprivkey, self.storage.clone())
+            .with_config(self.config.clone());
+        nm_builder.with_stop(self.stop.clone());
+        nm_builder.with_logger(self.logger.clone());
+        let node_manager = nm_builder.build().await?;
+
+        // create first node
+        let node_identity = node_manager.new_node().await?;
+
+        // replace the node manager
+        *lock = Some(node_manager);
+
+        // start sync
+        NodeManager::start_sync(self.node_manager.clone(), self.stop.clone());
+
+        Ok(node_identity)
+    }
+
+    /// Creates a node manager if we don't have one already.
+    pub async fn create_node_manager_if_needed(&self) -> Result<(), MutinyError> {
+        match self.create_node_manager().await {
+            Ok(_) => Ok(()),
+            Err(MutinyError::AlreadyRunning) => Ok(()), // this means we already have a node manager
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Removes the node manager from the wallet. If the node manager has any balances,
+    /// this function will fail.
+    pub async fn remove_node_manager(&self) -> Result<(), MutinyError> {
+        if self.is_safe_mode() {
+            return Err(MutinyError::NotRunning);
+        }
+
+        let mut lock = self.node_manager.write().await;
+        // if we don't have a node manager, return
+        if lock.is_none() {
+            return Ok(());
+        }
+
+        let node_manager = lock.as_ref().expect("should have a node manager");
+
+        if !node_manager.get_balance().await?.is_zero() {
+            return Err(MutinyError::InvalidArgumentsError);
+        }
```

Comment:
I think it should probably do more checks than just this. I feel like it should be as strict as possible. Stuff like no channels, no pending closes, perhaps as far as to say never having a channel. 

I can imagine what the purpose of `remove_node_manager` is, and I think we shouldn't ever take the node away from someone that has already used their node.

Reply:
Yeah I agree there, it is scary. My hope is that we can get people who have been fedimint only to click this, also I think it'd good to have for people who add a node manager just because they are exploring the app, they can get rid of it as well.

Do note, this function also calls `archive_node` which does the same check that we do for pruning monitors, if any monitors are not prunable, it will fail to archive. I'll add a comment so it is more clear when reading this function

Reply:
We could probably just call this immediately on everyone that has never had a node with channels.

Reply:
> We could probably just call this immediately on everyone that has never had a node with channels.

Added that and made the checks stronger so we should always unarchive the first node when coming back rather than always creating a new one.

Will start working on unit tests next.